app/google_calendar/google_calendar_api.py

```diff
@@ -0,0 +1,243 @@
+
+    def update_event(self, new_event: CalendarEvent, event_id: str) -> bool:
+        """Update an existing event in the calendar."""
+
+        try:
+            event_result = (
+                self.service.events()  # type: ignore
+                .update(
+                    calendarId=self.calendar_id,
+                    eventId=event_id,
+                    body={
+                        "summary": new_event.summary,
+                        "location": new_event.location,
+                        "description": new_event.description,
+                        "source": new_event.source,
+                        "extendedProperties": new_event.extendedProperties,
+                        "start": new_event.start,
+                        "end": new_event.end,
+                    },
+                )
+                .execute()
+            )
+
+            logger.info(f"Event updated in {self.calendar_id}")
+            logger.info("Updated event:")
+            logger.info(f"- id: {event_result['id']}")
+            logger.info(f"- summary: {event_result['summary']}")
+            logger.info(f"- starts at: {event_result['start']['dateTime']}")
+            logger.info(f"- ends at: {event_result['end']['dateTime']}")
+            logger.info(f"- location: {event_result['location']}")
+            logger.info(f"- description: {event_result['description']}")
+            logger.info(f"- extendedProperties: {event_result['extendedProperties']}")
+            return True
+
+        except Exception as e:
+            logger.error(f"Failed to update event {event_id} in {self.calendar_id}: {e}")
+            return False
+
+    def list_events(
+        self,
+        fetch_after_this_time: str,
+        max_events_to_fetch: int = settings.NUMBER_OF_EVENTS_TO_FETCH,
+    ) -> list[CalendarEvent]:
+        """List upcoming events in the calendar."""
+
+        logger.info(f"Getting all the events starting after time: {fetch_after_this_time}")
+        calendar_events: list[CalendarEvent] = []
+        try:
+            events_result = (
+                self.service.events()  # type: ignore
+                .list(
+                    calendarId=self.calendar_id,
+                    timeMin=fetch_after_this_time,
+                    singleEvents=True,
+                    maxResults=max_events_to_fetch,
+                    orderBy="startTime",
+                )
+                .execute()
+            )
+            events = events_result.get("items", [])
+            if not events:
+                logger.info(f"No upcoming events found in {self.calendar_id}.")
+            else:
+                logger.info(f"Found {len(events)} upcoming events")
+                # Convert events to CalendarEvent objects
+                calendar_events = [CalendarEvent(**event) for event in events]
+
+            return calendar_events
+
+        except Exception as e:
+            logger.error(f"Failed to list events in {self.calendar_id}: {e}")
```

Comment:
Why we need to catch the exception here and return empty array? why not just propagate the exception? (for all exceptions)

Reply:
updated